src/app/screens/TimesheetScreen/view/timesheetList/index.tsx

```diff
@@ -36,8 +36,8 @@ const TimesheetList = () => {
   const [isEditModalVisible, setIsEditModalVisible] = useState(false);
   const [editTimesheetData, setEditTimesheetData] = useState<Timesheet>();
   const [dateRange, setDateRange] = useState<TDateRange>({
-    startDate: startOfMonth,
-    endDate: todaysDate,
+    startDate: params?.startDate ? new Date(params.startDate) : startOfMonth,
+    endDate: params?.endDate ? new Date(params.endDate) : todaysDate,
```

Comment:
Here we are storing the Date object converting from a string. So we can directly get the Date object and then store it.